src/components/toast/toast.tsx

```diff
@@ -0,0 +1,205 @@
+  /**
+   * If the action type is button, this will be the text of the button:
+   */
+  @Prop() buttonText: string;
+  /**
+   * Time to close the toast in seconds
+   * 0 = no auto close (default value)
+   */
+  @Prop() duration = 0;
+  /**
+   * Define an action to the button toast. Can be one of:
+   * 'close', 'custom';
+   * if the action type is set to close, the button will close automatically.
+   * if the action type is set to custom, a function need to be passed when the toastButtonClick is emitted.
+   */
+  @Prop() buttonAction: ButtonActionType = "close";
+  /**
+   * Controls the open event of the component:
+   */
+  @Prop() show = false;
+    /**
+   * Controls the hide event of the component:
+   */
+  @Prop() hide = false;
+  /**
+   * Event used to execute some action when the action button on the toast is clicked
+   */
+  @Event() toastButtonClick!: EventEmitter;
+  /**
+   * Sends an event to be used when creating an action when clicking the toast button
+   */
+  private _buttonClickHandler = (event) => {
+    const element = event.path.find((item) => item.localName === "bds-toast");
+
+    if (this.buttonAction === "close") this.close(event);
+    else this.toastButtonClick.emit({ element });
+  };
+
+  /**
+   * Can be used outside to open the toast
+   */
+  @Method()
+  async create({
+    toastContainer,
+    toastElement,
+    actionType,
+    buttonAction,
+    buttonText,
+    icon,
+    toastText,
+    toastTitle,
+    variant,
+    duration,
+  }: CreateToastType) {
+    toastElement.actionType = actionType || 'button';
+    toastElement.buttonAction = buttonAction;
+    toastElement.buttonText = buttonText;
+    toastElement.icon = icon || 'info';
+    toastElement.toastText= toastText;
+    toastElement.toastTitle = toastTitle;
+    toastElement.variant = variant || 'system';
+    toastElement.duration = duration * 1000 || 0;
+
+    toastContainer.appendChild(toastElement);
+    toastElement.show = true;
+
+    if (toastElement.duration > 0) {
+      setTimeout(() => {
+        toastElement.hide = true;
+        setTimeout(() => {
+          toastElement.remove();
+        }, 400);
+      }, toastElement.duration);
+    }
+  }
+
+  /**
+   * Can be used outside the component to close the toast
+   */
+  @Method()
+  async close(event) {
+    const element = event.path.find((item) => item.localName === "bds-toast");
+    const tElement = await element.shadowRoot.childNodes[1];
+
+    tElement.classList.remove('show');
+    tElement.classList.add('hide');
+
+    setTimeout(() => {
+      element.remove();
+    }, 400);
+  }
+
+  render() {
+    return (
+      <div
+        class={{
+          toast: true,
+          [`toast--${this.variant}`]: true,
+          [`toast--action--${this.actionType}`]: true,
+          show: this.show,
+          hide: this.hide
+        }}
+      >
+        {this.icon && (
+          <bds-icon
+            theme="outline"
+            size="x-large"
+            color="#fff"
```

Comment:
Prefer `white`

Reply:
why?

Reply:
Bcs once I asked a developer which color is `#fff` and answered me black rsrs. So `white` is more meaningful.

Reply:
hahahaha, i've got your point, but i think this need to be a thing to be discussed, by now i think maybe it's better to continue to use the color code because we're using this way in all the components

Reply:
Ok, if it's the pattern, move on